node/test_forceios.js

```diff
@@ -0,0 +1,204 @@
+        + '      where appTypes are in: native, native_swift, react_native, hybrid_local, hybrid_remote\n'
+        + '\n'
+        + '  Clone https://github.com/FORK/SalesforceMobileSDK-iOS at branch BRANCH\n'
+        + '  Generate forceios package and deploys it to a temporary directory\n'
+        + '  Create and compile the application types selected\n'
+        , 'magenta');
+}
+
+//
+// Cleanup
+//
+function cleanup() {
+    log('Cleaning up temp dirs', 'green');
+    shelljs.rm('-rf', 'tmp');
+}
+
+//
+// Make temp dir and return its path
+//
+function mkTmpDir() {
+    var tmpDir = path.join('tmp', 'testforceios' + random(1000));
+    log('Making temp dir:' + tmpDir, 'green');
+    shelljs.mkdir('-p', tmpDir);
+    return tmpDir;
+}
+
+//
+// Clone iOS repo and return its path
+// 
+function cloneRepo(tmpDir, repoUrl, branch) {
+    log('Cloning ' + repoUrl + ' at ' + branch, 'green');
+    var repoDir = path.join(tmpDir, 'SalesforceMobileSDK-iOS');
+    shelljs.mkdir('-p', repoDir);
+    runProcess('git clone --branch ' + branch + ' --single-branch --depth 1 --recurse-submodules ' + repoUrl + ' ' + repoDir);
+    return repoDir;
+}
+
+//
+// Create and deploy forceios
+//
+function createDeployForceiosPackage(repoDir, tmpDir) {
+    log('Generating forceios package', 'green');
+    runProcess('ant -f ' + path.join(repoDir, 'build', 'build_npm.xml'));
+    runProcess('npm install --prefix ' + tmpDir + ' ' + path.join(repoDir, 'forceios-' + version + '.tgz'));
+}
+
+//
+// Create and compile non-hybrid app 
+//
+function createCompileApp(tmpDir, appType) {
+    log('Creating ' + appType + ' app ', 'green');
+    var appName = appType + 'App';
+    var appId = '3MVG9Iu66FKeHhINkB1l7xt7kR8czFcCTUhgoA8Ol2Ltf1eYHOU4SqQRSEitYFDUpqRWcoQ2.dBv_a1Dyu5xa';
+    var callbackUri = 'testsfdc:///mobilesdk/detect/oauth/done';
+    var forceiosPath = path.join(tmpDir, 'node_modules', '.bin', 'forceios');
+    var forceiosArgs = 'create '
+        + ' --apptype=' + appType
+        + ' --appname=' + appName
+        + ' --companyid=com.mycompany'
+        + ' --organization=MyCompany'
+        + ' --outputdir=' + tmpDir
+        + ' --appid=' + appId
+        + ' --callbackuri=' + callbackUri;
+    runProcess(forceiosPath + ' ' + forceiosArgs);
+
+    // Pointing to repoDir in Podfile
+    shelljs.sed('-i', /pod ('Salesforce.*')/g, 'pod $1, :path => \'../SalesforceMobileSDK-iOS\'', path.join(tmpDir, 'nativeApp', 'Podfile'));
+    shelljs.sed('-i', /pod ('Smart.*')/g, 'pod $1, :path => \'../SalesforceMobileSDK-iOS\'', path.join(tmpDir, 'nativeApp', 'Podfile'));
+
+    shelljs.pushd(path.join(tmpDir, appName));
+    runProcess('pod update');    
+    shelljs.popd();
+
+    var workspacePath = path.join(tmpDir, appName, appName + '.xcworkspace');
+    runProcess('xcodebuild -workspace ' + workspacePath + ' -scheme Pods-' + appName)
+}
+
+//
+// Create and compile hybrid app 
+//
+function createCompileHybridApp(tmpDir, appType) {
```

Comment:
For testing hybrid, perhaps we could run ```sed``` on the local ```forcedroid``` we're testing and add the branch and fork specifier to the line where ```CordovaPlugin``` repo is specified. This will enable us to test it as well as native. Only catch is that we have to manually update the ```CordovaPlugin``` repo on that branch before running the script, but I think that should be okay.